easybuild/tools/modules.py

```diff
@@ -237,14 +239,30 @@ def set_and_check_version(self):
         except (OSError), err:
             raise EasyBuildError("Failed to check version: %s", err)
 
-        if self.REQ_VERSION is None:
-            self.log.debug("No version requirement defined.")
+        if self.REQ_VERSION is None and self.MAX_VERSION is None:
+                self.log.debug("No version requirement defined.")
         else:
-            if StrictVersion(self.version) < StrictVersion(self.REQ_VERSION):
-                raise EasyBuildError("EasyBuild requires v%s >= v%s (no rc), found v%s",
-                                     self.__class__.__name__, self.REQ_VERSION, self.version)
+            if self.REQ_VERSION is None: # MAX_VERSION is not None
+                self.log.debug("No required minimum version defined.")
+                if StrictVersion(self.version) > StrictVersion(self.MAX_VERSION):
+                    raise EasyBuildError("EasyBuild requires v%s <= v%s (no rc), found v%s",
```

Comment:
You should probably drop the `no rc` part here, since using a release candidate of the highest allowed version is  older than that version, so it's OK to use it.